kythe/cxx/indexer/cxx/IndexerASTHooks.h

```diff
@@ -115,7 +118,14 @@ class IndexerASTVisitor : public RecursiveTypeVisitor<IndexerASTVisitor> {
         MarkedSources(&Sema, &Observer),
         ShouldStopIndexing(std::move(ShouldStopIndexing)),
         UsrByteSize(UsrByteSize),
-        DataflowEdges(EDE) {}
+        DataflowEdges(EDE) {
+    if (!TIEPP.empty()) {
+      re2::RE2::Options options;
+      options.set_never_capture(true);
+      TemplateInstanceExcludePathPattern =
+          std::make_unique<re2::RE2>(std::string(TIEPP), options);
```

Comment:
This silently ignores syntax errors in the regexp.  I think it makes more sense to accept a `std::unique_ptr<RE2>` (or `RE2*` or `re2::Set`) and CHECK that it is `ok()`.

Also, as an nit, there's no need to copy the `absl::string_view`; RE2 has a re2::StringPiece constructor so:

`absl::make_unique<re2::RE2>({TIEPP.data(), TIEPP.size()}, options);`

The above should work both internally and externally.